src/core/server/saved_objects/service/lib/repository.test.js

```diff
@@ -2477,6 +2477,33 @@ describe('SavedObjectsRepository', () => {
         expect(client.search).not.toHaveBeenCalled();
       });
 
+      it(`throws when namespaces is an empty array`, async () => {
+        await expect(
+          savedObjectsRepository.find({ type: 'foo', namespaces: [] })
+        ).rejects.toThrowError('options.namespaces cannot be an empty array');
+        expect(client.search).not.toHaveBeenCalled();
+      });
```

Comment:
We are still allowing `namespaces: undefined`, right?

Reply:
Yes, as long as `typeToNamespacesMap` is not defined, you can use `namespaces: undefined` as a valid argument (to search in the Default space).